foundationdb/src/main/java/site/ycsb/db/foundationdb/FoundationDBClient.java

```diff
@@ -182,6 +199,11 @@ private Status convTupleToMap(Tuple tuple, Set<String> fields, Map<String, ByteI
   private void batchInsert(int dbIndex) {
     try {
       dbs[dbIndex].run(tr -> {
+          if (setTransactionTrace && Math.random()<transactionTraceFraction) {
+            tr.options().setDebugTransactionIdentifier(debugTransactionIdentifier);
+            tr.options().setLogTransaction();
+            tr.options().setServerRequestTracing();
+          }
```

Comment:
It would be cleaner to put these 5 lines in a private function called like **maybeSetTransactionTrace()**. @sears what's your opinion?

Reply:
Agreed.  Just before the other copies of this stanza, we also have copy/pasted `if (setPriorityBatch) ...` blocks.  Those could also go in the helper.

Unless I'm misreading something the `batch` in `setPriorityBatch` and `batchInsert` are unrelated concepts.  The `if(setPriorityBatch) { ... }` block probably should be included in batchInsert too.  So, if this gets moved to a private helper, include the setPriorityBatch logic, and maybe call it something like `setupTransaction()` or something.

Reply:
Yes, let me put the three transaction trace options into a helper function. However, I don't think it is good idea to put every option into this single helper function, since different operations may have different settings. batchInsert is a case, and scan is another case, and there is no guarantee of options for any future changes.
Another personal flavor is that I would like to put the "if" outside the helper function, since it is clear to read and has the same format as when we set setPriorityBatch option.